internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -60,25 +87,30 @@ var secretSpec = &schema.Schema{
 	Description: "Spec for the kubernetes secret",
 	Optional:    true,
 	MaxItems:    1,
+	MinItems:    1,
 	Elem: &schema.Resource{
 		Schema: map[string]*schema.Schema{
```

Comment:
nit: Please add any validations/constraints that have been specified in the swagger file. This applies to the entire schema.

Reply:
added PTAL